crates/uv-resolver/src/resolver/environment.rs

```diff
@@ -0,0 +1,420 @@
+use std::sync::Arc;
+
+use uv_pep508::{MarkerEnvironment, MarkerTree};
+use uv_pypi_types::ResolverMarkerEnvironment;
+
+use crate::requires_python::RequiresPythonRange;
+use crate::resolver::ForkState;
+use crate::PythonRequirement;
+use crate::ResolveError;
+
+/// Represents the environment that dependencies must satisfy in a resolution.
```

Comment:
I'd phrase the other round, we're building a resolution that works for the specified environment(s), ignoring dependency edge outside of it.

Reply:
Hmmm I see. It's a little tortured, but I re-phrased it like so:

```
/// Represents one or more marker environments for a resolution.
///
/// Dependencies outside of the marker environments represented by this value
/// are ignored for that particular resolution.
///
/// In normal "pip"-style resolution, one resolver environment corresponds to
/// precisely one marker environment. In universal resolution, multiple marker
/// environments may be specified via a PEP 508 marker expression. In either
/// case, as mentioned above, dependencies not in these marker environments are
/// ignored for the corresponding resolution.
```